### PR TITLE
Right click menu

### PR DESCRIPTION
**Pull request by iqmulator**

### CODE REVIEW

File: src/dotnet/Chat.UI.Blazor/Components/RightClickChatMenu/RightClickMenuDelete.razor [hunk 1]
```diff
@@ -0,0 +1,24 @@
+@namespace ActualChat.Chat.UI.Blazor.Components
+@using BlazorContextMenu
+
+<Item OnClick="@(_ => OnDelete(Chat.Id))" CssClass="context-menu-item warning-item">
```
Comment (alexis-kochetov) [inline src/dotnet/Chat.UI.Blazor/Components/RightClickChatMenu/RightClickMenuDelete.razor:4]: Rename method to `Delete` or replace ad-hoc method with `OnDeleteClick` handler method

**Review by alexis-kochetov — CHANGES_REQUESTED** [review]

**Review by crui3er — CHANGES_REQUESTED** [review]

**Comment by iqmulator** [comment]
image

**Comment by iqmulator** [comment]
image
image

File: src/dotnet/Chat.UI.Blazor/Components/ChatView/Items/ChatMessage.razor [hunk 2]
```diff
@@ -91,7 +91,7 @@
 
     @if (!message.IsQuote) {
         <CascadingValue Value="@message">
-            <ContextMenu
+            <CustomContextMenu
```
Comment (FrolYo) [inline src/dotnet/Chat.UI.Blazor/Components/ChatView/Items/ChatMessage.razor:94]: prefix `custom` looks useless

**Review by FrolYo — CHANGES_REQUESTED** [review]

File: src/dotnet/Chat.UI.Blazor/Components/RightClickChatMenu/RightClickMenuPin.razor [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    protected override ComputedState<bool>.Options GetStateOptions()
+        => new() { UpdateDelayer = UpdateDelayer.MinDelay };
+
+    protected override async Task<bool> ComputeState(CancellationToken cancellationToken) {
+        var pinnedChatIds = await ChatUI.PinnedChatIds.Use(cancellationToken).ConfigureAwait(false);
+        return pinnedChatIds.Contains(Chat.Id);
+    }
+
+    private Task Toggle() {
+        var pinnedChatIds = ChatUI.PinnedChatIds.Value;
```
Comment (FrolYo) [inline src/dotnet/Chat.UI.Blazor/Components/RightClickChatMenu/RightClickMenuPin.razor:53]: @alexyakunin @crui3er
is it fine here to call `Value`? Or we should call `Use()` instead?

File: src/dotnet/Chat.UI.Blazor/Components/RightClickChatMenu/RightClickMenuPlayback.razor [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+            InitialValue = Model.None,
+            UpdateDelayer = UpdateDelayer.ZeroDelay,
+        };
+
+    protected override async Task<Model> ComputeState(CancellationToken cancellationToken) {
+        var isPlayingActive = await ChatUI.IsPlaying.Use(cancellationToken).ConfigureAwait(false);
+        var chatPlaybackState = await ChatPlayers.ChatPlaybackState.Use(cancellationToken).ConfigureAwait(false);
+        var isPlayingHistorical = chatPlaybackState is HistoricalChatPlaybackState;
+        if (!isPlayingActive)
+            return new Model() {
```
Comment (FrolYo) [inline src/dotnet/Chat.UI.Blazor/Components/RightClickChatMenu/RightClickMenuPlayback.razor:61]: `new ()`?

File: src/dotnet/Chat.UI.Blazor/Components/RightClickChatMenu/RightClickMenuRecord.razor [hunk 5]
```diff
@@ -0,0 +1,55 @@
+@namespace ActualChat.Chat.UI.Blazor.Components
+@using BlazorContextMenu
+@inherits ComputedStateComponent<Symbol>
+
+@{
+    var isRecording = RecordingChatId == Chat.Id;
+    var cls = "context-menu-item important-item " + (isRecording ? "on" : "off");
```
Comment (FrolYo) [inline src/dotnet/Chat.UI.Blazor/Components/RightClickChatMenu/RightClickMenuRecord.razor:7]: `context-menu-item important-item` part looks constant. How about moving it into markup?

**Review by FrolYo — CHANGES_REQUESTED** [review]

File: src/dotnet/Chat.UI.Blazor/Components/RightClickChatMenu/RightClickMenuRecord.razor (hunk 5 above)
Comment (iqmulator) [reply src/dotnet/Chat.UI.Blazor/Components/RightClickChatMenu/RightClickMenuRecord.razor:7]: important-item class highlights red items like delete

File: src/dotnet/Chat.UI.Blazor/Components/ChatView/Items/ChatMessage.razor (hunk 2 above)
Comment (iqmulator) [reply src/dotnet/Chat.UI.Blazor/Components/ChatView/Items/ChatMessage.razor:94]: Renamed to context-menu-old

**Review by FrolYo — APPROVED** [review]

**Pull request merged**